docs/tutorials/FlowTutorial.md

```diff
@@ -0,0 +1,1581 @@
+the launch clock arrives earlier than the capture clock.
+
+Refer to the following `gcd` design on repair_timing with fast and slow corners.
+
+Refer built-in example [here](../../test/gcd_sky130hd_fast_slow.tcl)
+
+Run the follwoing commands in termina:
+```
+cd ../test/
+openroad
+source gcd_sky130hd_fast_slow.tcl
+```
+
+The resulting `worst slack`, `TNS` follows
+
+```
+report_worst_slack -min -digits 3
+worst slack 0.321
+report_worst_slack -max -digits 3
+worst slack -16.005
+report_tns -digits 3
+tns -529.496
+```
+
+#### Fixing Setup Violations
+
+To fix setup timing path violations command use `repair_timing -setup`
+
+Refer to the following built-in example to learn more about fixing setup timing
+violations.
+
+Refer built-in example [here](../../src/rsz/test/repair_setup4.tcl)
+
+Run the following Tcl command in terminal:
+
+```
+cd ../../src/rsz/test/
+openroad -gui
+```
+
+Copy and paste following commands in Tcl commands section of GUI.
+
+```
+define_corners fast slow
+read_liberty -corner slow Nangate45/Nangate45_slow.lib
+read_liberty -corner fast Nangate45/Nangate45_fast.lib
+read_lef Nangate45/Nangate45.lef
+read_def repair_setup1.def
+create_clock -period 0.3 clk
+set_wire_rc -layer metal3
+estimate_parasitics -placement
+report_checks -fields input -digits 3
+```
+
+View the generated timing report with slack violation.
+
+```
+Startpoint: r1 (rising edge-triggered flip-flop clocked by clk)
+Endpoint: r2 (rising edge-triggered flip-flop clocked by clk)
+Path Group: clk
+Path Type: max
+Corner: slow
+
+   Delay     Time   Description
+-----------------------------------------------------------
+   0.000    0.000   clock clk (rise edge)
+   0.000    0.000   clock network delay (ideal)
+   0.000    0.000 ^ r1/CK (DFF_X1)
+   0.835    0.835 ^ r1/Q (DFF_X1)
+   0.001    0.836 ^ u1/A (BUF_X1)
+   0.196    1.032 ^ u1/Z (BUF_X1)
+   0.001    1.033 ^ u2/A (BUF_X1)
+   0.121    1.154 ^ u2/Z (BUF_X1)
+   0.001    1.155 ^ u3/A (BUF_X1)
+   0.118    1.273 ^ u3/Z (BUF_X1)
+   0.001    1.275 ^ u4/A (BUF_X1)
+   0.118    1.393 ^ u4/Z (BUF_X1)
+   0.001    1.394 ^ u5/A (BUF_X1)
+   0.367    1.761 ^ u5/Z (BUF_X1)
+   0.048    1.809 ^ r2/D (DFF_X1)
+            1.809   data arrival time
+
+   0.300    0.300   clock clk (rise edge)
+   0.000    0.300   clock network delay (ideal)
+   0.000    0.300   clock reconvergence pessimism
+            0.300 ^ r2/CK (DFF_X1)
+  -0.155    0.145   library setup time
+            0.145   data required time
+-----------------------------------------------------------
+            0.145   data required time
+           -1.809   data arrival time
+-----------------------------------------------------------
+           -1.664   slack (VIOLATED)
+
+```
+
+To fix this setup violation use below command:
+
+```
+repair_timing -setup
+```
+
+The log as follows:
+
+```
+[INFO RSZ-0040] Inserted 4 buffers.
+[INFO RSZ-0041] Resized 16 instances.
+[WARNING RSZ-0062] Unable to repair all setup violations.

```

Comment:
What should the user do if they encounter this? You should probably show how to clean up the design to get proper timing.

Reply:
@vijayank88 Suggest to the user to try lowering the clock frequency and re-run  to resolve the setup violation in this case.